python/tests/api/logger/test_segments.py

```diff
@@ -426,3 +427,27 @@ def test_segment_merge_different_columns() -> None:
             assert segmented_view._columns["A"] is not None
             assert segmented_view._columns["B"] is not None
         assert segmented_view._columns["A"]._metrics["cardinality"].estimate == pytest.approx(1.0)
+
+
+def test_segment_with_nans() -> None:
+    df = pd.DataFrame({"col_1": [1, 2, 3, 4, 5, 6], "col_nan": [True, True, None, None, np.nan, math.nan]})
+    column_segments = segment_on_column("col_nan")
```

Comment:
We also need a test that segments on more than one column to reproduce the multi-column segmentation bug.

Reply:
this is done  later in this same test, starting in line 442

Reply:
you are right! sorry missed that.